### ISSUE TITLE
Black background after minimizing and expanding the app

### ISSUE THREAD

**Issue by Holofox** [Baseflow/flutter-permission-handler#363]
## 🐛 Bug Report

If you call a dialog with a request for permissions and minimize the application, having previously opened it again, then the dialog will have a black background.

### Expected behavior

After returning to the application again, the UI should render correctly.

### Reproduction steps

You can view the reproduction of the bug here: https://github.com/flutter/flutter/issues/58002

### Configuration

**Version:** 5.0.1+1

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android

**Comment by pinpong** [comment]
I can confirm the issue, this happens to me with an In-App billing dialog from google.

**Comment by psycura** [comment]
I have same issues on LocationPermissions dialog

**Comment by psycura** [comment]
Any updates?

**Comment by aloev** [comment]
Is happening to me now, anyone with the fix?

**Comment by ReveredMachine** [comment]
I faced the same issue when doing the following
- Open App-Settings with openAppSettings()
- Do some permission changes
- Navigate to my App
--> The App hangs in the Splash-Screen

When doing the above without changing any permisskions it semms to be working.

**Comment by DOMO-Liu** [comment]
I have the same problem, the state of the lock screen also occurs.

**Comment by ildarsharafutdinov** [comment]
I do confirm the issue with share plugin.

Steps to reproduce:
- run share plugin example https://github.com/flutter/plugins/tree/share-v0.6.5+4/packages/share/example
- input text and tap share button
- block and unblock phone screen

Share dialog is ok, however background is black. Once share dialog is closed, the app appears on the screen.
The same is true for photo/camera permissions request dialog.

**Comment by JDDV** [comment]
This is not a permission handler bug, this looks like a bug in flutter or Android it self.

I tested this using the standard incremental counter app and using different packages that were able to request a permission.
When the permission dialog showed up, I pressed the 'menu' button, putitng the app in the background, and re-opened the app. With every package I tested it with the screen turned black. The link provided by @ildarsharafutdinov shows a nice and simple solution to fix this problem.

This issue will be closed for now, if you still have questions regarding this problem please feel free to start a new issue.
